### PR TITLE
Add a smoke test

### PR DESCRIPTION
**Pull request by twolodzko**
Basic smoke test

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import tensorflow as tf
+import pytest
+
+
+@pytest.fixture(scope="session")
+def dataset():
+    mnist = tf.keras.datasets.mnist
+    (x_train, y_train), (x_test, y_test) = mnist.load_data()
+    x_train, x_test = x_train / 255.0, x_test / 255.0
+    yield (x_train, y_train), (x_test, y_test)
```
Comment (kshitij12345) [inline tests/conftest.py:11]: Would it make sense to use `return`? I think `yield` would make sense if there is some clean-up code later.

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            tf.keras.layers.Dense(10),
+        ]
+    )
+    loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
+    model.compile(
+        optimizer="adam",
+        loss=loss_fn,
+        metrics=["accuracy"],
+    )
+    yield model
```
Comment (kshitij12345) [inline tests/conftest.py:30]: yield -> return?

File: tests/test_e2e.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import pytest
+
+from neptune_tensorflow_keras.impl import NeptuneCallback
+
+
+try:
+    # neptune-client=0.9.0+ package structure
+    from neptune.new import init_run
+except ImportError:
```
Comment (kshitij12345) [inline tests/test_e2e.py:14]: Are we planning to run this test with older version of the client?

File: tests/test_e2e.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+def test_smoke(dataset, model):
+    run = init_run()
+    callback = NeptuneCallback(run=run, base_namespace='metrics')
+
+    (x_train, y_train), (x_test, y_test) = dataset
+
+    model.fit(x_train, y_train, epochs=5, callbacks=[callback])
+
+    assert run.exists("metrics")
+    assert run.exists("metrics/train")
```
Comment (kshitij12345) [inline tests/test_e2e.py:28]: Are these two necessary given that we are checking `metrics/train/*` below.

**Review by kshitij12345 — COMMENTED** [review]
Looks good. Have a few inline questions.

Thank you :)

File: tests/test_e2e.py (hunk 3 above)
Comment (twolodzko) [reply tests/test_e2e.py:14]: We use this template in other repositories, so I assume that we want to be able to do this.

File: tests/test_e2e.py (hunk 4 above)
Comment (twolodzko) [reply tests/test_e2e.py:28]: No.

File: tests/conftest.py (hunk 2 above)
Comment (twolodzko) [reply tests/conftest.py]: I took this from Prophet integration tests, but you are correct return is fine here.

**Pull request closed without merging**